plugins/module_utils/foreman_helper.py

```diff
@@ -90,6 +92,32 @@ def inner(self, *args, **kwargs):
     return decor
 
 
+def _check_patch_needed(fixed_version=None, plugins=None):
+    """
+    Decorator to check whether a specific apidoc patch is required.
+
+    :param fixed_version: The version of Foreman the API bug was fixed.
+    :type fixed_version: str, optional
+    :param plugins: Which plugins are required for this patch.
+    :type plugins: list, optional
+    """
+
+    def decor(f):
+        @wraps(f)
+        def inner(self, *args, **kwargs):
+            if plugins is not None:
+                for plugin in plugins:
+                    if not self.has_plugin(plugin):
+                        return
```

Comment:
If you think it's easier to read:
```suggestion
            if plugins is not None and not all(self.has_plugin(plugin) for plugin in plugins):
                return
```